---
title: "PMC Survey Data Analysis"
author: "Mark Ziemann & Kaumadi Wijesooriya"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    fig_width: 7
    fig_height: 5
theme: cosmo
---

Source: https://github.com/markziemann/SurveyEnrichmentMethods

## Intro

Here we are performing an analysis of some gene expression data to demonstrate the difference between ORA and FCS methods and to highlight the differences caused by improper background gene set use.

```{r,begin}

library("getDEE2") 
library("DESeq2")
library("clusterProfiler")
library("mitch")
library("kableExtra")
library("eulerr")

```

## Get expression data

I'm using some RNA-seq data looking at the effect of hyperglycemia on hepatocytes.

```{r,getdata}

mdat<-getDEE2Metadata("hsapiens")

#SRP128998
#samplesheet<-mdat[which(mdat$GEO_series=="GSE109140"),]
samplesheet <- mdat[grep("SRP128998",mdat$SRP_accession),]
samplesheet<-samplesheet[order(samplesheet$SRR_accession),]
samplesheet$HG<-as.factor(c(1,1,1,1,1,1,0,0,0,0,0,0))
samplesheet$VPA<-as.factor(c(0,0,0,1,1,1,0,0,0,1,1,1))

# samplesheets
s1 <- subset(samplesheet,VPA==0)

w<-getDEE2("hsapiens",samplesheet$SRR_accession,metadata=mdat,legacy = TRUE)
x<-Tx2Gene(w)
x<-x$Tx2Gene

# save the genetable for later
gt<-w$GeneInfo[,1,drop=FALSE]
gt$accession<-rownames(gt)
head(gt)

# counts 
x1<-x[,which(colnames(x) %in% s1$SRR_accession)]

# filter out lowly expressed genes
x1<-x1[which(rowSums(x1)/ncol(x1)>=(10)),]

#run DESeq2 for LG vs HG
y <- DESeqDataSetFromMatrix(countData = round(x1), colData = s1, design = ~ HG)
y <- DESeq(y)
LGvHG <- results(y)
LGvHG<-as.data.frame(LGvHG[order(LGvHG$pvalue),])
rownames(LGvHG)<-sapply(strsplit(rownames(LGvHG),"\\."),"[[",1)
head(LGvHG)

```

## Gene sets from Reactome

```{r,reactome}

download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip", 
    destfile="ReactomePathways.gmt.zip")
unzip("ReactomePathways.gmt.zip")
genesets<-gmt_import("ReactomePathways.gmt")

```

## FCS with Mitch

```{r,mitch1}
library(kableExtra)

m <- mitch_import(LGvHG,DEtype = "DEseq2", geneTable = gt)
head(m)
mres <- mitch_calc(m,genesets = genesets)
head(mres$enrichment_result,50)
m_up <- subset(mres$enrichment_result,p.adjustANOVA<0.05 & s.dist > 0)[,1]
m_dn <- subset(mres$enrichment_result,p.adjustANOVA<0.05 & s.dist < 0)[,1]
length(m_up)
length(m_dn)

```

## ORA with clusterprofiler

```{r,cp1}


genesets2 <- read.gmt("ReactomePathways.gmt")
head(genesets2)

head(LGvHG)

LGvHG_up <- rownames(subset(LGvHG,log2FoldChange>0,padj<0.05))
LGvHG_up <- unique(gt[which(rownames(gt) %in% LGvHG_up),1])
head(LGvHG_up)

LGvHG_dn <- rownames(subset(LGvHG,log2FoldChange<0,padj<0.05))
LGvHG_dn <- unique(gt[which(rownames(gt) %in% LGvHG_dn),1])
head(LGvHG_dn)

LGvHG_bg <- rownames(LGvHG)
LGvHG_bg <- unique(gt[which(rownames(gt) %in% LGvHG_bg),1])
length(LGvHG_bg)
head(LGvHG_bg)

c_up <- enricher(gene = LGvHG_up, universe = LGvHG_bg, TERM2GENE = genesets2)
c_up <- head(c_up,1000)
head(c_up,20)
c_up <- rownames(subset(c_up, p.adjust < 0.05))
       
c_dn <- enricher(gene = LGvHG_dn, universe = LGvHG_bg, TERM2GENE = genesets2)
c_dn <- head(c_dn,1000)
head(c_up,20)
c_dn <- rownames(subset(c_dn, p.adjust < 0.05))

```

## ORA with clusterprofiler with whole genome background list

```{r,cp1}

LGvHG_bg <- w$GeneInfo$GeneSymbol
length(LGvHG_bg)
head(LGvHG_bg)

f_up <- enricher(gene = LGvHG_up, universe = LGvHG_bg, TERM2GENE = genesets2)
f_up <- head(f_up,1000)
head(f_up,20)
f_up <- rownames(subset(f_up, p.adjust < 0.05))
       
f_dn <- enricher(gene = LGvHG_dn, universe = LGvHG_bg, TERM2GENE = genesets2)
f_dn <- head(f_dn,1000)
head(f_up,20)
f_dn <- rownames(subset(f_dn, p.adjust < 0.05))

```
## Venn diagram comparison

```{r,venn1}

v1 <- list("FCS up"=m_up, "FCS dn"=m_dn,
           "ORA up"=c_up,"ORA dn"=c_dn)
  
plot(euler(v1),quantities = TRUE)

v2 <- list("ORA up"=c_up,"ORA dn"=c_dn, 
           "ORA* up"=f_up,"ORA* dn"=f_dn )

plot(euler(v2),quantities = TRUE)

v3 <- list("FCS up"=m_up, "FCS dn"=m_dn,
           "ORA up"=c_up,"ORA dn"=c_dn, 
           "ORA* up"=f_up,"ORA* dn"=f_dn )

plot(euler(v3),quantities = TRUE)

```



## Session information

```{r,session}

sessionInfo()

```