---
title: "PMC Survey Analysis Part 1"
author: "Mark Ziemann & Kaumadi Wijesooriya"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    fig_width: 7
    fig_height: 5
theme: cosmo
---
  
## Intro

Here we are performing an analysis of 200 articles which is randomly selected from 1500 PMC articles.
These articles were examined independently by two team members and the results were compared.
Any inconsistencies were resolved.

The code shown here was used to generate Figure 2 of the manuscript.

```{r,begin}

knitr::opts_chunk$set(fig.width=7, fig.height=5) 

library("wordcloud")
library("RColorBrewer")
library("wordcloud2")
library("reutils")
library("XML")
library("kableExtra")
library("Biobase")
library("vioplot")

```

## Overview of included and excluded analyses

```{r,exclude1}

x <- read.table("../data/QC-analysis2.tsv",header=TRUE,fill=TRUE,sep="\t")
head(x)
colnames(x)
dim(x)

exclude <- subset(x,x$GS.version=="EXCLUDE")
nrow(exclude)
length(unique(exclude$Pubmed.Central.ID))

x <- subset(x,x$GS.version!="EXCLUDE")
nrow(x)
length(unique(x$Pubmed.Central.ID))

```
## Journal 

```{r,Journal1}

journal <- x$Journal
journal_split <- strsplit(journal,", ")
journal <- unlist(journal_split)
res <- table(journal)
res <- res[order(res)]
length(res)
res

par(mar=c(1,1,1,1))
#names(res) <- gsub("Gene expression array","RNA array",names(res))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(5,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "Journal", xlim=c(0,17))
grid()

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Journal", xlim=c(0,165))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 7, label = res2, pos = 3, cex = 1, col = "black")

dir.create("images")
png("images/journals2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Journal", xlim=c(0,170))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 10, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/journals2.pdf",width=4,height=4)
par(mar=c(5,9,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Journal", xlim=c(0,180))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 15, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```


## Omics type

```{r,omics1}

omics <- x$Omics.type
omics_split <- strsplit(omics,", ")
omics <- unlist(omics_split)
res <- table(omics)
res <- res[order(res)]
length(res)
res

par(mar=c(1,1,1,1))
names(res) <- gsub("Gene expression array","RNA array",names(res))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(5,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "Omics type", xlim=c(0,100))
grid()

names(res) <- gsub("RNA array","Gene expression array",names(res))

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Omics type", xlim=c(0,100))
text(y = (1:length(res2)*1.2)-1.2  , x = res2+5, label = res2, pos = 3, cex = 1, col = "black")

png("images/omics2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Omics type", xlim=c(0,100))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 6, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/omics2.pdf",width=4,height=4)
par(mar=c(5,10,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Omics type", xlim=c(0,110))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 10, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```

## Organism

```{r,org1}

org <- x$Organism
org_split <- strsplit(org,", ")
org <- unlist(org_split)
res <- table(org)
res <- res[order(res)]
length(res)
res

par(mar=c(1,1,1,1))
names(res) <- gsub("Homo sapiens","human",names(res))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"), scale=c(4,.5))


par(mar=c(5,12,3,5))
names(res) <- gsub("human","Homo sapiens",names(res))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "Organism", xlim=c(0,200))
grid()

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Organism", xlim=c(0,200))
text(y = (1:length(res2)*1.2)-1.2  , x = res2+7, label = res2, pos = 3, cex = 1, col = "black")

png("images/organisms2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Organism", xlim=c(0,200))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 15, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/organisms2.pdf",width=4,height=4)
par(mar=c(5,11,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Organism", xlim=c(0,200))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 25, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```

## Gene set library

```{r,gsl1}

GSL <-x$Gene.set.library
GSL_split <- strsplit(GSL,", ")
GSL <- unlist(GSL_split)
res <- table(GSL)
res <- res[order(res)]
length(res)
which(names(res)=="Not stated")/sum(res)*100
res

par(mar=c(1,1,1,1))
names(res) <- gsub("Homo sapiens","human",names(res))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(5,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "Gene set library", xlim=c(0,140))
grid()

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set library", xlim=c(0,140))
text(y = (1:length(res2)*1.2)-1.2  , x = res2+7, label = res2, pos = 3, cex = 1, col = "black")

png("images/genesetlib2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set library", xlim=c(0,150))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 15, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/genesetlib2.pdf",width=4,height=4)
par(mar=c(5,11,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set library", xlim=c(0,150))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 16, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```

## Gene set version

```{r,gsv1}

GSV <-x$GS.version
res <- table(GSV)
res
res[1]/sum(res)*100

par(mar=c(14,12,3,10))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set version defined", xlim=c(0,250))
text(y = (1:length(res)*1.2) - 0.75 , x = res+15, label = res, pos = 3, cex = 1, col = "black")


png("images/genesetvers2.png",width=300,height=150)
par(mar=c(5,5,3,3))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set version defined", xlim=c(0,250))
text(y = (1:length(res)*1.2)-1.2  , x = res + 20, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/genesetvers2.pdf",width=3,height=2)
par(mar=c(5,4,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene set version defined", xlim=c(0,260))
text(y = (1:length(res)*1.2)-1.2  , x = res + 25, label = res, pos = 3, cex = 1, col = "black")
dev.off()


```

## Statistical test used

```{r,test1}

test <-x$Statistical.test.used
test <- strsplit(test,", ")
test <- unlist(test)
res <- table(test)
res <- res[order(res)]
res[which(names(res)=="Not stated")] / sum(res) * 100
length(res)
res


par(mar=c(1,1,1,1))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(5,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "Test used", xlim=c(0,150))
grid()

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Test used", xlim=c(0,150))
text(y = (1:length(res2)*1.2)-1.1  , x = res2+10, label = res2, pos = 3, cex = 1, col = "black")

png("images/stattest2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Test used", xlim=c(0,160))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 15, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/stattest2.pdf",width=4,height=4)
par(mar=c(5,11,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Test used", xlim=c(0,170))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 20, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```

## FDR Correction

```{r,fdr1}

fdr <-x$FDR.Correction
fdr <- strsplit(fdr,", ")
fdr <- unlist(fdr)
res <- table(fdr)
res <- res[order(res)]
res[which(names(res)!="Yes")]/sum(res)*100
res

par(mar=c(1,1,1,5))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(10,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "FDR correction performed", xlim=c(0,150))
text(y = (1:length(res)*1.2) - 0.8 , x = res+10, label = res, pos = 3, cex = 1, col = "black")

png("images/fdr2.png",width=300,height=200)
par(mar=c(5,8,3,3))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "FDR correction performed", xlim=c(0,160))
text(y = (1:length(res)*1.2)-1.2  , x = res + 15, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/fdr2.pdf",width=3,height=3)
par(mar=c(5,5,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "FDR correction performed", xlim=c(0,160))
text(y = (1:length(res)*1.2)-1.0  , x = res + 15, label = res, pos = 3, cex = 1, col = "black")
dev.off()


```


## App used

```{r,app1}

App <-x$App.used
App_split <- strsplit(App,", ")
App <- unlist(App_split)
res <- table(App)
res <- res[order(res)]
res[which(names(res)=="Not stated")]/sum(res)*100
length(res)
res

par(mar=c(1,1,1,1))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(5,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 0.7, xlab="no. analyses",
        main = "App used", xlim=c(0,60))
grid()

other <- sum(res[1:(nrow(res)-10)])
res2 <- c(other,tail(res,9))
names(res2)[1] <- "Other"
par(mar=c(5,12,3,5))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App used", xlim=c(0,80))
text(y = (1:length(res2)*1.2)-1.1  , x = res2+5, label = res2, pos = 3, cex = 1, col = "black")

png("images/app2.png",width=400,height=300)
par(mar=c(5,12,3,3))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App used", xlim=c(0,80))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 5, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/app2.pdf",width=4,height=4)
par(mar=c(5,12,3,2))
barplot(res2,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App used", xlim=c(0,80))
text(y = (1:length(res2)*1.2)-1.2  , x = res2 + 10, label = res2, pos = 3, cex = 1, col = "black")
dev.off()

```

## App version

```{r,appv1}

APV <-x$App.Version
res <- table(APV)
res
res[1]/sum(res)*100

par(mar=c(14,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App version defined", xlim=c(0,200))
text(y = (1:length(res)*1.2) - 0.75 , x = res+15, label = res, pos = 3, cex = 1, col = "black")

png("images/appvers2.png",width=300,height=150)
par(mar=c(5,5,3,3))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App version defined", xlim=c(0,200))
text(y = (1:length(res)*1.2)-1.2  , x = res + 15, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/appvers2.pdf",width=3,height=2)
par(mar=c(5,4,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "App version defined", xlim=c(0,200))
text(y = (1:length(res)*1.2)-1.2  , x = res + 19, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```

## Code available

```{r,code1}

code <-x$Code.availability
res <- table(code)
res
res[1]/sum(res)*100

par(mar=c(14,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Code availability", xlim=c(0,50))
text(y = (1:length(res)*1.2) - 0.75 , x = res+2, label = res, pos = 3, cex = 1, col = "black")

png("images/code2.png",width=300,height=150)
par(mar=c(5,5,3,3))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Code availability", xlim=c(0,50))
text(y = (1:length(res)*1.2)-1.2  , x = res + 3, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/code2.pdf",width=3,height=2)
par(mar=c(5,4,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Code availability", xlim=c(0,55))
text(y = (1:length(res)*1.2)-1.2  , x = res + 5, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```

## Background gene set

```{r,background1}

BG <-x$Background.gene.set
BG <- gsub("Yes","Yes, correct",BG)
res <- table(BG)
res
sum(res[which(names(res)!="Yes")])/sum(res)*100

par(mar=c(10,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Background list specification", xlim=c(0,200))
text(y = (1:length(res)*1.2) - 0.85 , x = res+10, label = res, pos = 3, cex = 1, col = "black")

png("images/bg2.png",width=300,height=200)
par(mar=c(5,9,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Background list specified", xlim=c(0,220))
text(y = (1:length(res)*1.2)-1.1  , x = res + 18, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/bg2.pdf",width=4,height=3)
par(mar=c(5,10,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Background list specified", xlim=c(0,220))
text(y = (1:length(res)*1.2)-1.1  , x = res + 20, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```

## Gene lists provided

```{r,genelists1}

GL <-x$Gene.lists.provided
res <- table(GL)
res
sum(res[which(names(res)!="Yes")])/sum(res)*100

par(mar=c(14,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene lists provided", xlim=c(0,160))
text(y = (1:length(res)*1.2) - 0.75 , x = res+8, label = res, pos = 3, cex = 1, col = "black")

png("images/genelists2.png",width=300,height=150)
par(mar=c(5,5,3,3))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene lists provided", xlim=c(0,160))
text(y = (1:length(res)*1.2)-1.2  , x = res + 9, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/genelists2.pdf",width=3,height=2)
par(mar=c(5,4,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Gene lists provided", xlim=c(0,170))
text(y = (1:length(res)*1.2)-1.25  , x = res + 15, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```

## Methodological flaws

```{r,assumptions1}

ok <- nrow(subset(x,Assumptions.violated=="No"))
ok
bad <- nrow(subset(x,Assumptions.violated!="No"))
bad
ok/sum(bad,ok)*100

ass <-x$Assumptions.violated
ass <- gsub("^No$","None",ass)
ass <- strsplit(ass,", ")
ass <- unlist(ass)
res <- table(ass)
res <- res[order(res)]
res

par(mar=c(1,1,1,1))
wordcloud(words = names(res), freq = res, min.freq = 1, 
          max.words=200, random.order=FALSE, rot.per=0.35, 
          colors=brewer.pal(8, "Dark2"))

par(mar=c(8,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Methodological flaws", xlim=c(0,200))
text(y = (1:length(res)*1.2) - 0.9 , x = res+8, label = res, pos = 3, cex = 1, col = "black")

png("images/assumptions2.png",width=400,height=250)
par(mar=c(5,12,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Methodological flaws", xlim=c(0,220))
text(y = (1:length(res)*1.2)-1.1  , x = res + 18, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/assumptions2.pdf",width=4,height=3)
par(mar=c(5,11,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Methodological flaws", xlim=c(0,230))
text(y = (1:length(res)*1.2)-1.2  , x = res + 25, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```


## Separated or combined

```{r,combined}

comb <-x$Separated.up.and.down
res <- table(comb)
res
sum(res[which(names(res)!="Separated")])/sum(res)*100
res <- res[which(names(res)!="-")]


par(mar=c(10,12,3,5))
barplot(tail(res,20),horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Separated or combined ORA (differential expression)", xlim=c(0,90))
grid()
text(y = (1:length(res)*1.2) - 0.9 , x = res+5, label = res, pos = 3, cex = 1, col = "black")

png("images/comb2.png",width=300,height=200)
par(mar=c(5,8,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Separated or combined ORA", xlim=c(0,100))
text(y = (1:length(res)*1.2)-1.1  , x = res + 5, label = res, pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/comb2.pdf",width=3,height=2.5)
par(mar=c(5,5,3,2))
barplot(res,horiz=TRUE,las=1,cex.names = 1, xlab="no. analyses",
        main = "Separated or combined ORA", xlim=c(0,100))
text(y = (1:length(res)*1.2)-1.1  , x = res + 10, label = res, pos = 3, cex = 1, col = "black")
dev.off()

```

## GSEA parameters

```{r,gseaparam}

param <-x$Parameters
param <- param[which(param!="-")]
param <- gsub(" ","",param)
str(param)

rankspec <- sapply(strsplit(param,","),"[[",1)
rankspec <- gsub("rank=","",rankspec)
rankspec <- table(rankspec)
rankspec

algospec <- sapply(strsplit(param,","),"[[",2)
algospec <- gsub("algo=","",algospec)
algospec <- table(algospec)
algospec

weightspec <- sapply(strsplit(param,","),"[[",3)
weightspec <- gsub("weight=","",weightspec)
weightspec <- table(weightspec)
weightspec

par(mar=c(10,12,3,4))
params <- rbind(rankspec,algospec,weightspec)
rownames(params) <- c("rank method", "test type", "weight type")
colnames(params) <- c("not stated", "stated")
barplot(t(params),horiz=TRUE,las=1,xlim=c(0,30),legend = colnames(params),
        main="GSEA parameter reporting", xlab="no. analyses",
        args.legend = list(x = "topleft", inset = c(0.05, 0.1)))
text(y = (1:nrow(params)*1.2) - 0.7 , x = params[,1]-3, label = params[,1], pos = 3, cex = 1, col = "white")
text(y = (1:nrow(params)*1.2) - 0.7 , x = params[,1]+2, label = params[,2], pos = 3, cex = 1, col = "black")

png("images/gseaparam2.png",width=350,height=250)
par(mar=c(5,8,3,2))
params <- rbind(rankspec,algospec,weightspec)
rownames(params) <- c("rank method", "test type", "weight type")
colnames(params) <- c("not stated", "stated")
barplot(t(params),horiz=TRUE,las=1,xlim=c(0,30),legend = colnames(params),
        main="GSEA parameter reporting", xlab="no. analyses",
        args.legend = list(x = "topleft", inset = c(0.05, 0.1)))
text(y = (1:nrow(params)*1.2) - 0.7 , x = params[,1]-3, label = params[,1], pos = 3, cex = 1, col = "white")
text(y = (1:nrow(params)*1.2) - 0.7 , x = params[,1]+2, label = params[,2], pos = 3, cex = 1, col = "black")
dev.off()

pdf("images/gseaparam2.pdf",width=4.5,height=3)
par(mar=c(5,6,3,2))
barplot(t(params),horiz=TRUE,las=1,xlim=c(0,30),legend = colnames(params),
        main="GSEA parameter reporting", xlab="no. analyses",
        args.legend = list(x = "topleft", bg="white",inset = c(0.05, 0.1)))
text(y = (1:nrow(params)*1.2) - 0.8 , x = params[,1]-3, label = params[,1], pos = 3, cex = 1, col = "white")
text(y = (1:nrow(params)*1.2) - 0.8 , x = params[,1]+2, label = params[,2], pos = 3, cex = 1, col = "black")
dev.off()

```

## Session information

```{r,session}

sessionInfo()

```
