# Proposed title: A survey of functional enrichment research practices

### Supervisory team

Dr Mark Ziemann will be primary supervisor and we will investigate candidates for a secondary supervisor.

### Background

Gene set enrichment analyses are a widely used and powerful class of analytical tools for summarising omics data.
Despite this, when performed improperly it can lead to biases and incorrect conclusions [1](Timmons et al, 2015).
Dr Ziemann has observed that sometimes, such enrichment analyses are poorly described and suffer from methodological flaws.
It remains unknown as to how prevalent these problems are in the contemporary literature.

### Aim and hypothesis

The purpose of this study is to perform a survey of the use of contemporary enrichment analysis methods and the level of documentation in the literature.
The aim is two-fold.
Firstly we will provide the research community with a survey of research methods which will include the popularity of certain tools along with the frequency that methodological or reporting flaws.
Secondly, we will compose checklist of best practices for performing and reporting functional enrichment analysis in the style of Minimum information about a microarray experiment (MIAME)[2](Brazma et al, 2001).
By doing this, we hope to shed light on how researchers can avoid common pitfalls in enrichment analysis.

### Research approach
We will collate a list of papers in PubMed Central (https://www.ncbi.nlm.nih.gov/pmc/) published in 2019 that have “enrichment analysis” and related terms. 
With each of these papers we will collect information:

* Pubmed Central ID

* Journal name

* Type of omics data

* Species

* Gene set library used

* Statistical test used

* Whether p-values were FDR corrected

* Software package used

* Versions of software and functional classifications were reported

* Code availability

* Whether a background gene set was used

* Whether assumptions are correctly applied

* Other notes (for example if there is no documentation of enrichment analysis).

Specifically we will be determining the presence of poor analytical practices in enrichment analysis in the contemporary literature.
We will be using R to analyze this data.

Target Journal: PLoS Computational Biology, Genome Research

### Project support and logistics

Dr Ziemann has in excess of 10 years genomics experience and will train the candidate to undertake these analyses.
There will be weekly progress meetings with Dr Ziemann and monthly meetings with the entire supervisory team.
In the period before the start of the project, the candidate and supervisor will meet every two weeks to discuss the project.

The supervisory team will make all reasonable endeavours to support the candidate during the research including adjusting the contact time if necessary.
The supervisory team will promptly respond to correspondence on the next banking day.
The supervisory team will actively assist to solve technical issues such as computer code bugs to keep the project moving.
The candidate will contribute research via GitHub including code and thesis writing. The project is compatible with the two trimester time-frame of Masters projects at Deakin.

### Expectations

For the duration of the project, the candidate is expected to work for 30 to 40 hours per week on the project.
That time commitment includes research, reading, coding, writing, administrative tasks, relevant self-education and group activities.
The candidate will draw up a timeline for thesis completion including milestones for the completion of the research and writing of each chapter.
The candidate is expected to stick to their own timeline.
The candidate is entitled to two weeks time off from the project that can be taken at any time so long as the supervision team is notified at least one week beforehand.
The candidate will participate in the group activities including weekly group meeting.
The candidate is expected  to return any computer equipment to the supervision team if they have borrowed any.

### References

1. Timmons JA, Szkop KJ, Gallagher IJ. Multiple sources of bias confound functional enrichment analysis of global -omics data. Genome Biol. 2015;16(1):186. Published 2015 Sep 7. doi:10.1186/s13059-015-0761-7

2. Brazma A, Hingamp P, Quackenbush J, et al. Minimum information about a microarray experiment (MIAME)-toward standards for microarray data. Nat Genet. 2001;29(4):365-371. doi:10.1038/ng1201-365

### Related reading

3. Tipney H, Hunter L. An introduction to effective use of enrichment analysis software. Hum Genomics. 2010;4(3):202-206. doi:10.1186/1479-7364-4-3-202

4. Khatri P, Sirota M, Butte AJ. Ten years of pathway analysis: current approaches and outstanding challenges. PLoS Comput Biol. 2012;8(2):e1002375. doi:10.1371/journal.pcbi.1002375

5. Conesa A, Madrigal P, Tarazona S, et al. A survey of best practices for RNA-seq data analysis [published correction appears in Genome Biol. 2016;17(1):181]. Genome Biol. 2016;17:13. Published 2016 Jan 26. doi:10.1186/s13059-016-0881-8
